packages/core/src/browser/preferences/preference-configurations.ts

```diff
@@ -16,16 +16,22 @@
 
 import { injectable, inject, named, interfaces } from 'inversify';
 import URI from '../../common/uri';
+import { EnvVariablesServer } from '../../common/env-variables';
 import { ContributionProvider, bindContributionProvider } from '../../common/contribution-provider';
 
 export const PreferenceConfiguration = Symbol('PreferenceConfiguration');
 export interface PreferenceConfiguration {
     name: string;
 }
+export const DATA_FOLDER_NAME = Symbol('DataFolderName');
 
 export function bindPreferenceConfigurations(bind: interfaces.Bind): void {
     bindContributionProvider(bind, PreferenceConfiguration);
     bind(PreferenceConfigurations).toSelf().inSingletonScope();
+    bind(DATA_FOLDER_NAME).toDynamicValue(async ctx => {
```

Comment:
does this really work? I'd assume it binds to a `Promise<string>` which wouldn't help us.

Reply:
No it's not. That the issue of this PR. I tried to follow Anton's advice from [here](https://github.com/eclipse-theia/theia/pull/6650#discussion_r359206948) and [there](https://github.com/eclipse-theia/theia/pull/6650#discussion_r358754429), but it didn't work.
Maybe you have an idea how we can resolve the problem?

Reply:
@akosyakov  @svenefftinge generally we want to have sync getter for a backend property from frontend which seems to be impossible (or I don't know how to do it).
So, we need to change approach. Another option could be defining this value at compile time, say via environment variable and the use, for example, [webpack plugin](https://webpack.js.org/plugins/define-plugin/).

Reply:
@mmorhun I've just looked again I thought you changed `PreferenceService.initializeProviders` method, but it was `FoldersPreferencesProvider.updateProviders`. Sorry for confusion. It should be fine to change `PreferencConfigurations.getPaths` to be async there.